cas-server-core/src/main/java/org/jasig/cas/CentralAuthenticationServiceImpl.java

```diff
@@ -256,8 +257,8 @@ public ServiceTicket grantServiceTicket(
             }
             ticketGrantingTicket.getSupplementalAuthentications().add(currentAuthentication);
         }
-        
-        if (!registeredService.isSsoEnabled() && currentAuthentication == null) {
+
+        if (currentAuthentication == null && !registeredService.getAuthorizationStrategy().isServiceAuthorizedForSso(service)) {

```

Comment:
You're right. The service object that is passed right now is only complementary and in the default impl of the authz strategy, only is used for logging purposes. The problem is that the authz strategy, even though it's associated with a registered service has no way to accessing its properties which is why this pull sort of cheats :) but I can certainly clean up the API to remove the recursive passing of the extra parameter.